### DISCUSSION TITLE
The value of the budget was miscalculated

### DISCUSSION THREAD

**Discussion by Darren-Li** [pymc-labs/pymc-marketing#1263] [General]
https://www.pymc-marketing.io/en/stable/notebooks/mmm/mmm_case_study.html#budget-optimization

![image](https://github.com/user-attachments/assets/06a31ef5-325a-46f3-8042-9499d71703be)
the budget should be the overall spend for all channels in one week. so the code should be 
`budget = X_test[channel_columns].mean(axis=0).sum()`
but not the everage spend for one channel

also, the code for the chat in the image is wrong as well, And, obviously, it doesn't match the picture below.

The same goes for the next one.
![image](https://github.com/user-attachments/assets/ff35856b-5627-421b-a880-b2feb1881e08)

```python
mean_spend_per_period_test = (
    # X_test[channel_columns].sum(axis=0).div(num_periods * len(channel_columns))
    X_test[channel_columns].mean(axis=0)
)
```

**Comment by williambdean** [comment]
Hi, Thanks for bringing to our attention!
Is this also present in the latest docs? https://www.pymc-marketing.io/en/latest/notebooks/mmm/mmm_case_study.html#budget-optimization

**Reply by Darren-Li** [reply]
Thank you for your prompt reply:)

From your shared page(dev verison), I also have some points, you can double check again.

1. For code snippet below. `all_budget_per_channel.div(num_periods * len(channel_columns))` should be changed to `all_budget_per_channel.div(num_periods )`
![image](https://github.com/user-attachments/assets/805c9d0a-9ee4-47c8-96fb-0702ac0c5413)

2. The same for those snippets below. It should not be divided by `len(channel columns)`
![image](https://github.com/user-attachments/assets/03816f8c-e7d5-470b-a324-91cd8732df45)
![image](https://github.com/user-attachments/assets/06a814c3-6db0-42c6-905f-e4e2b39f166d)

3. I got this error when run the below code `AttributeError: 'MMM' object has no attribute 'optimize_budget'`, maybe it is new function in an upcoming update. But one of my doubts is `budget=per_channel_weekly_budget,`, is it right, or it should be `all_budget #Total budget for the test set` ?
![image](https://github.com/user-attachments/assets/ea664188-9ee4-4653-8486-dc034723706a)

**Reply by Darren-Li** [reply]
Can I find out when a version update for MMM is coming?

**Reply by williambdean** [reply]
We are working on an official release. However, you can pip install the main branch from github

**Reply by williambdean** [reply]
CC @cetagostini

**Reply by Darren-Li** [reply]
> We are working on an official release. However, you can pip install the main branch from github

I will wait for the release, I am afraid of making a mistake, ha ha ha, I am not very familiar with these operations.
